hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/service/KeyDeletingService.java

```diff
@@ -269,7 +281,15 @@ private void processSnapshotDeepClean(int delCount)
             String snapshotBucketKey = dbBucketKey + OzoneConsts.OM_KEY_PREFIX;
             SnapshotInfo previousSnapshot = getPreviousActiveSnapshot(
                 currSnapInfo, snapChainManager, omSnapshotManager);
+            SnapshotInfo previousToPrevSnapshot = null;
+
+            if (previousSnapshot != null) {
```

Comment:
nit: it would be clean if you use `Optional`. It is a perfect case to use optional.